### PR TITLE
Review sorting

### PR DESCRIPTION
**Pull request by filipemir**

### CODE REVIEW

File: spec/features/user_sees_reviews_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    end
+
+    check_order(review1, review2)
+    check_order(review3, review2)
+    check_order(review3, review1)
+    check_order(review1, review4)
+    check_order(review6, review4)
+    check_order(review5, review4)
+  end
+end
```
Comment (filipemir) [inline spec/features/user_sees_reviews_spec.rb]: No it's not

File: app/models/review.rb [hunk 2]
```diff
@@ -16,8 +16,10 @@ class Review < ActiveRecord::Base
   validates :elevator_id, presence: true
   validates :rating, presence: true
 
-  def total_votes
+  def count_votes
     votes = Vote.where(review_id: id)
-    votes.inject(0) { |a, vote| a + vote.vote }
+    self.total_votes = votes.inject(0) { |a, vote| a + vote.vote }
```
Comment (justinhuynh) [inline app/models/review.rb]: you shouldn't need `self` here

File: spec/features/user_sees_reviews_spec.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  let!(:review6) { FactoryGirl.create(:review, elevator: elevator, total_votes: 0) }
+
+  scenario 'user sees reviews sorted by vote number and review date' do
+    visit elevators_path
+    click_on 'Cool Elevator'
+
+    def check_order(earlier_review, later_review)
+      earlier_content = 'id="review-' + earlier_review.id.to_s + '"'
+      later_content = 'id="review-' + later_review.id.to_s + '"'
+      expect(earlier_content).to appear_before(later_content)
```
Comment (justinhuynh) [inline spec/features/user_sees_reviews_spec.rb]: fancy! but you should move your method definitions outside of the scenario. try creating a file `helpers.rb` in your `spec` folder next your `factories.rb`, and stick this method definition in `module Helpers`

[example here](https://www.relishapp.com/rspec/rspec-core/docs/helper-methods/define-helper-methods-in-a-module)

Include `config.include Helpers` in your `rails_helper` file, inside the `RSpec` config block

File: spec/features/user_sees_reviews_spec.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+RSpec::Matchers.define :appear_before do |later_content|
```
Comment (justinhuynh) [inline spec/features/user_sees_reviews_spec.rb]: Again - neat stuff. Try putting this in your `rails_helper` file instead

Comment (filipemir) [inline spec/features/user_sees_reviews_spec.rb]: Makes sense. Added.

File: app/models/review.rb (hunk 2 above)
Comment (filipemir) [inline app/models/review.rb]: Not sure I can tell you why but we definitely need it. It doesn't work without it. Happy to show you!

Comment (justinhuynh) [inline app/models/review.rb]: that's fine.

**Pull request merged**